research/build-reverse-dependencies/index.md

```diff
@@ -0,0 +1,98 @@
+---
+title: Build reverse dependencies in Packit
+authors: mmassari
+---
```

Comment:
Second part of the general comment about the supported project structures. Where are the packages to be rebuilt

- Monorepo
  - Easy to control what triggers build chains
  - Probably already the discussion of a different research/issue
- Monorepo with external references, i.e. point a package to a git repo and use the `packit.yaml` inside there
  - Good control of what packages to rebuild
  - Requires dependent project to have packit integration
- Completely external reference
  - Source and build environment can be anything, dist-git, copr project or packit project
  - Can dynamically query all dependencies using `fedrq` (only in case of dist-git source)

Reply:
I don't think there is a need for changing the project structure but I am not sure I am following you.
The `reverse-dependencies` should be dynamically rebuilt (and also discovered if we don't go in the way of statically defining them).
This is done more easily when we work with Koji and the Fedora ecosystem where all the dependencies should already be satisfied, https://koschei.fedoraproject.org/ already does it and you don't need to specify anything.
If we decide to use COPR and enable every upstream repo we should probably find a way to describe in the new `Job` the reverse dependencies (I would say statically) and a way to build them in COPR (there could be many approaches for this also the `action` can be used).
For me this is a "last step", an enhancement in the new `Job` but if I understand it correctly you would benefit from it because not all your packages are released in Fedora?

Reply:
> but if I understand it correctly you would benefit from it because not all your packages are released in Fedora?

Yes, this is for pre-release/pre-packaging. Referencing a git-repo under packit would be sufficient for that.

> If we decide to use COPR and enable every upstream repo

The main supported upstream model here sould be to build in copr with dist-git source (probably from `fedrq`).

> I don't think there is a need for changing the project structure 

I was thinking of the structure of monorepo, just because it would be more natural to navigate a dependency tree and it would share much of the scheduling logic that is needed for monorepo dependencies.

Indeed an alternative would be to have it in Jobs, but I'm afraid complex dependencies would become harder to manage.

These would be another upstream model for when some dependencies are not yet packaged.